third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1389,51 +1416,58 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 					return opErr
 				}
 			}
-		}
-
-		if d.HasChange(prefix + ".alias_ip_range") {
-			rereadFingerprint := false
 
-			// Alias IP ranges cannot be updated; they must be removed and then added.
-			if len(instNetworkInterface.AliasIpRanges) > 0 {
-				ni := &computeBeta.NetworkInterface{
-					Fingerprint:     instNetworkInterface.Fingerprint,
-					ForceSendFields: []string{"AliasIpRanges"},
-				}
-				op, err := config.clientComputeBeta.Instances.UpdateNetworkInterface(project, zone, instance.Name, networkName, ni).Do()
-				if err != nil {
-					return errwrap.Wrapf("Error removing alias_ip_range: {{err}}", err)
-				}
-				opErr := computeOperationWaitTime(config, op, project, "updating alias ip ranges", d.Timeout(schema.TimeoutUpdate))
-				if opErr != nil {
-					return opErr
-				}
-				rereadFingerprint = true
+			//re-read fingerprint
+			instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
+			if err != nil {
+				return err
 			}
+			instNetworkInterface = instance.NetworkInterfaces[i]
+		}
 
-			ranges := d.Get(prefix + ".alias_ip_range").([]interface{})
-			if len(ranges) > 0 {
-				if rereadFingerprint {
+		if !updateDurringStop {
+			if d.HasChange(prefix + ".alias_ip_range") {
+				// Alias IP ranges cannot be updated; they must be removed and then added.
+				// unless you are changing subnetwork/network
+				if len(instNetworkInterface.AliasIpRanges) > 0 {
+					ni := &computeBeta.NetworkInterface{
+						Fingerprint:     instNetworkInterface.Fingerprint,
+						ForceSendFields: []string{"AliasIpRanges"},
+					}
+					op, err := config.clientComputeBeta.Instances.UpdateNetworkInterface(project, zone, instance.Name, networkName, ni).Do()
+					if err != nil {
+						return errwrap.Wrapf("Error removing alias_ip_range: {{err}}", err)
+					}
+					opErr := computeOperationWaitTime(config, op, project, "updating alias ip ranges", d.Timeout(schema.TimeoutUpdate))
+					if opErr != nil {
+						return opErr
+					}
+					//re-read fingerprint
 					instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
 					if err != nil {
 						return err
 					}
 					instNetworkInterface = instance.NetworkInterfaces[i]
 				}
-				ni := &computeBeta.NetworkInterface{
-					AliasIpRanges: expandAliasIpRanges(ranges),
-					Fingerprint:   instNetworkInterface.Fingerprint,
-				}
-				op, err := config.clientComputeBeta.Instances.UpdateNetworkInterface(project, zone, instance.Name, networkName, ni).Do()
-				if err != nil {
-					return errwrap.Wrapf("Error adding alias_ip_range: {{err}}", err)
-				}
-				opErr := computeOperationWaitTime(config, op, project, "updating alias ip ranges", d.Timeout(schema.TimeoutUpdate))
-				if opErr != nil {
-					return opErr
-				}
 			}
 		}
+
+		networkInterface.Fingerprint = instNetworkInterface.Fingerprint
+		networkInterface.NetworkIP = ""
+		networkInterface.AccessConfigs = nil
```

Comment:
Can you add a comment explaining why we have to set NetworkIP and AccessConfigs to empty here?